src/gql/server/Buttons.tsx

```diff
@@ -178,57 +209,53 @@ export function Slew({ label, disabled, className }: { label: string; disabled:
       variables={variables}
-      className={className}
-      label={label}
-      disabled={disabled || !selectedTarget.id || loading}
+      {...props}
+      loading={props.loading || loading}
+      disabled={props.disabled || !selectedTarget.id}
     />
   );
 }
 
-// OIWFS
-const OIWFS_MUTATION = graphql(`
-  mutation setOiwfsTarget(
-    $id: TargetId!
-    $name: NonEmptyString!
-    $ra: HmsString
-    $dec: DmsString
-    $epoch: EpochString
-    $wavelength: PosBigDecimal
-  ) {
-    oiwfsTarget(
-      target: {
-        id: $id
-        name: $name
-        sidereal: { ra: { hms: $ra }, dec: { dms: $dec }, epoch: $epoch }
-        wavelength: { nanometers: $wavelength }
-      }
-    ) {
-      result
-    }
-  }
-`);
+// TODO: is this needed anywhere? I'm not sure what it is
+// export function Oiwfs({ state, className, ...props }: { state: MechSystemState | undefined } & ButtonProps) {
+//   const { oiTargets } = useTargets().data;
+//   const configuration = useConfiguration().data?.configuration;
+//   const selectedTarget = oiTargets.find((t) => t.pk === configuration?.selectedOiTarget);
 
-export function Oiwfs({ label, disabled, className = '' }: { label: string; disabled: boolean; className?: string }) {
-  const { oiTargets } = useTargets().data;
-  const configuration = useConfiguration().data?.configuration;
-  const selectedTarget = oiTargets.find((t) => t.pk === configuration?.selectedOiTarget);
+//   const classes = clsx(className, classNameForState(state, true));
+//   return selectedTarget ? (
+//     <MutationButton
+//       mutation={OIWFS_MUTATION}
+//       variables={{
+//         id: selectedTarget.id!,
+//         name: selectedTarget.name,
+//         ra: selectedTarget.ra?.hms,
+//         dec: selectedTarget.dec?.dms,
+//         epoch: selectedTarget.epoch,
+//         wavelength: 400,
+//       }}
+//       {...props}
+//       className={classes}
+//       disabled={props.disabled ?? !selectedTarget.name}
+//     />
+//   ) : (
+//     <Button {...props} className={classes} />
+//   );
+// }
 
-  return selectedTarget ? (
-    <MutationButton
-      mutation={OIWFS_MUTATION}
-      variables={{
-        id: selectedTarget.id!,
-        name: selectedTarget.name,
-        ra: selectedTarget.ra?.hms,
-        dec: selectedTarget.dec?.dms,
-        epoch: selectedTarget.epoch,
-        wavelength: 400,
-      }}
-      className={className}
-      label={label}
-      disabled={disabled || !selectedTarget.name}
-    />
-  ) : (
-    <Button className={className} disabled={disabled} label={label} />
-  );
+/**
+ *
+ * @param state State of the subsystem
+ * @param usedSubsystem If the subsystem is being used. Everything is always used, except the guiders (PWFS1, PWFS2, OIWFS, AOWFS), which are optional. Which one is used is given by the guide environment. But right now, we can use only OIWFS
+ */
+function classNameForState(state: MechSystemState | undefined, usedSubsystem: boolean) {
```

Comment:
I'm not fully confident the logic of this function is correct. Please review carefully :)

Reply:
I think we need to discuss a bit better the devices state from the server.
What you have now it is right, but in the beginning I did not have a way to know if the device was `not active` or in an error state.
The states written in the original documentation is here https://docs.google.com/document/d/1WPqImRgELyQ6mO00SbcWdD7F949VCoEPXgQfMNSj590/edit#heading=h.7hl1bnj05r1b on page 14.